phdu.go

```diff
@@ -69,24 +72,13 @@ func NewPrimaryHDU(hdr *Header) (Image, error) {
 			Value:   len(hdr.Axes()),
 			Comment: "number of data axes",
 		})
-	}
-
-	if len(hdr.Axes()) >= 1 {
-		if _, ok := keys["NAXIS1"]; !ok {
-			cards = append(cards, Card{
-				Name:    "NAXIS1",
-				Value:   hdr.Axes()[0],
-				Comment: "length of data axis 1",
-			})
-		}
-	}
-
-	if len(hdr.Axes()) >= 2 {
-		if _, ok := keys["NAXIS2"]; !ok {
+		nax := len(hdr.Axes())
+		for i := 0; i < nax; i++ {
+			axis := strconv.Itoa(i + 1) // index from 0, hdu starts at NAXIS1
```

Comment:
you should test whether that data card key already exist in `keys` before adding it to `cards`.